src/main/java/seedu/hirehub/model/ModelManager.java

```diff
@@ -271,6 +272,26 @@ public void replaceApplications(Job target, Job editedJob) {
         applicationList.setApplications(applications);
     }
 
+    @Override
+    public int countRemainingVacancy(String jobTitle) {
+        int countAccepted = 0;
+        int countVacancy = 0;
+        Job jobToFind = new Job(jobTitle, "", 1);
+        for (Job job : jobList) {
+            if (job.isSameJob(jobToFind)) {
+                countVacancy = job.getVacancy();
+            }
+        }
```

Comment:
This part should be extracted out into a function to allow for early termination once the job is found